jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java

```diff
@@ -33,7 +33,9 @@
   private static boolean initialized = false;
   private static final String nativePath = "../../lib/";
   private static final String nativeResourcePath = "/lib/";
-  private static final String[] libNames = new String[]{"xgboost4j"};
+  private static final String[] libNames = System.getProperty("os.name").equals("Linux")
+          ? new String[]{"xgboost4j", "c", "stdc++"}
```

Comment:
This doesn't work just yet, because the library is linked against e.g. libstdc++.so.6 and not libstdc++.so. Therefore the linker is not able to load the override.